File: src/main/java/uk/gov/justice/hmpps/probationteams/dto/LocalDeliveryUnitDto.java
package uk.gov.justice.hmpps.probationteams.dto;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.*;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;


@ApiModel(description = "Local Delivery Unit")
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@ToString
public class LocalDeliveryUnitDto {

    @ApiModelProperty(required = true, value = "Local Delivery Unit code", position = 1, example = "NO2")
    @NotBlank
    private String code;

    @ApiModelProperty(required = true, value = "Functional Mailbox", position = 2, example = "a@b.com")
    @Email
    private String functionalMailBox;
}

File: src/main/java/uk/gov/justice/hmpps/probationteams/dto/LocalDeliveryUnitDto.kt
package uk.gov.justice.hmpps.probationteams.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank

data class LocalDeliveryUnitDto @JsonCreator constructor(
  @Schema(description = "Probation Area code", example = "NO2", required = true)
  @JsonProperty("probationAreaCode")
  val probationAreaCode: @NotBlank String,

  @Schema(description = "Local Delivery Unit code", example = "NO2SUK", required = true)
  @JsonProperty("localDeliveryUnitCode")
  val localDeliveryUnitCode: @NotBlank String,

  @Schema(description = "Functional Mailbox", example = "a@b.com", required = true)
  @JsonProperty("functionalMailbox")
  val functionalMailbox: @Email String? = null,

  @JsonProperty("probationTeams")
  val probationTeams: Map<String, ProbationTeamDto>? = mapOf(),
)
